GLOB sdist-make: /workdir/defusedxml-0.7.1/setup.py
cpythonlibtest-unit-test-tests create: /workdir/defusedxml-0.7.1/.tox/cpythonlibtest-unit-test-tests
cpythonlibtest-unit-test-tests installdeps: mock, pytest
cpythonlibtest-unit-test-tests inst: /workdir/defusedxml-0.7.1/.tox/.tmp/package/1/defusedxml-0.7.1.zip
cpythonlibtest-unit-test-tests installed: defusedxml @ file:///workdir/defusedxml-0.7.1/.tox/.tmp/package/1/defusedxml-0.7.1.zip#sha256=b471b87d93f05e214b4a9040ce6d4ba60c7a9fde8389b2880ddf536f2cf1bd0e,exceptiongroup==1.1.3,iniconfig==2.0.0,mock==5.1.0,packaging==23.2,pluggy==1.3.0,pytest==7.4.3,tomli==2.0.1
cpythonlibtest-unit-test-tests run-test-pre: PYTHONHASHSEED='3982981673'
cpythonlibtest-unit-test-tests run-test: commands[0] | pytest -v --tb=native --junitxml /workdir/results/defusedxml/0.7.1/1/cpython-test-results.xml
============================= test session starts ==============================
platform linux -- Python 3.10.8, pytest-7.4.3, pluggy-1.3.0 -- /workdir/defusedxml-0.7.1/.tox/cpythonlibtest-unit-test-tests/bin/python
cachedir: .tox/cpythonlibtest-unit-test-tests/.pytest_cache
rootdir: /workdir/defusedxml-0.7.1
configfile: tox.ini
collecting ... collected 91 items

tests.py::BaseTests::test_allow_expansion FAILED                         [  1%]
tests.py::BaseTests::test_dtd_forbidden FAILED                           [  2%]
tests.py::BaseTests::test_dtd_with_external_ref FAILED                   [  3%]
tests.py::BaseTests::test_entities_forbidden FAILED                      [  4%]
tests.py::BaseTests::test_entity_cycle FAILED                            [  5%]
tests.py::BaseTests::test_external_file_ref FAILED                       [  6%]
tests.py::BaseTests::test_external_ref FAILED                            [  7%]
tests.py::BaseTests::test_simple_parse FAILED                            [  8%]
tests.py::BaseTests::test_simple_parse_ns FAILED                         [  9%]
tests.py::TestDefusedElementTree::test_aliases PASSED                    [ 10%]
tests.py::TestDefusedElementTree::test_allow_expansion PASSED            [ 12%]
tests.py::TestDefusedElementTree::test_dtd_forbidden PASSED              [ 13%]
tests.py::TestDefusedElementTree::test_dtd_with_external_ref PASSED      [ 14%]
tests.py::TestDefusedElementTree::test_entities_forbidden PASSED         [ 15%]
tests.py::TestDefusedElementTree::test_entity_cycle PASSED               [ 16%]
tests.py::TestDefusedElementTree::test_external_file_ref PASSED          [ 17%]
tests.py::TestDefusedElementTree::test_external_ref PASSED               [ 18%]
tests.py::TestDefusedElementTree::test_html_arg FAILED                   [ 19%]
tests.py::TestDefusedElementTree::test_import_order PASSED               [ 20%]
tests.py::TestDefusedElementTree::test_orig_parseerror PASSED            [ 21%]
tests.py::TestDefusedElementTree::test_simple_parse PASSED               [ 23%]
tests.py::TestDefusedElementTree::test_simple_parse_ns PASSED            [ 24%]
tests.py::TestDefusedcElementTree::test_aliases PASSED                   [ 25%]
tests.py::TestDefusedcElementTree::test_allow_expansion PASSED           [ 26%]
tests.py::TestDefusedcElementTree::test_celementtree_warnings PASSED     [ 27%]
tests.py::TestDefusedcElementTree::test_dtd_forbidden PASSED             [ 28%]
tests.py::TestDefusedcElementTree::test_dtd_with_external_ref PASSED     [ 29%]
tests.py::TestDefusedcElementTree::test_entities_forbidden PASSED        [ 30%]
tests.py::TestDefusedcElementTree::test_entity_cycle PASSED              [ 31%]
tests.py::TestDefusedcElementTree::test_external_file_ref PASSED         [ 32%]
tests.py::TestDefusedcElementTree::test_external_ref PASSED              [ 34%]
tests.py::TestDefusedcElementTree::test_html_arg FAILED                  [ 35%]
tests.py::TestDefusedcElementTree::test_import_order PASSED              [ 36%]
tests.py::TestDefusedcElementTree::test_orig_parseerror PASSED           [ 37%]
tests.py::TestDefusedcElementTree::test_simple_parse PASSED              [ 38%]
tests.py::TestDefusedcElementTree::test_simple_parse_ns PASSED           [ 39%]
tests.py::TestDefusedMinidom::test_allow_expansion PASSED                [ 40%]
tests.py::TestDefusedMinidom::test_dtd_forbidden PASSED                  [ 41%]
tests.py::TestDefusedMinidom::test_dtd_with_external_ref PASSED          [ 42%]
tests.py::TestDefusedMinidom::test_entities_forbidden PASSED             [ 43%]
tests.py::TestDefusedMinidom::test_entity_cycle PASSED                   [ 45%]
tests.py::TestDefusedMinidom::test_external_file_ref PASSED              [ 46%]
tests.py::TestDefusedMinidom::test_external_ref PASSED                   [ 47%]
tests.py::TestDefusedMinidom::test_simple_parse PASSED                   [ 48%]
tests.py::TestDefusedMinidom::test_simple_parse_ns PASSED                [ 49%]
tests.py::TestDefusedMinidomWithParser::test_allow_expansion PASSED      [ 50%]
tests.py::TestDefusedMinidomWithParser::test_dtd_forbidden PASSED        [ 51%]
tests.py::TestDefusedMinidomWithParser::test_dtd_with_external_ref PASSED [ 52%]
tests.py::TestDefusedMinidomWithParser::test_entities_forbidden PASSED   [ 53%]
tests.py::TestDefusedMinidomWithParser::test_entity_cycle PASSED         [ 54%]
tests.py::TestDefusedMinidomWithParser::test_external_file_ref PASSED    [ 56%]
tests.py::TestDefusedMinidomWithParser::test_external_ref PASSED         [ 57%]
tests.py::TestDefusedMinidomWithParser::test_simple_parse PASSED         [ 58%]
tests.py::TestDefusedMinidomWithParser::test_simple_parse_ns PASSED      [ 59%]
tests.py::TestDefusedPulldom::test_allow_expansion PASSED                [ 60%]
tests.py::TestDefusedPulldom::test_dtd_forbidden PASSED                  [ 61%]
tests.py::TestDefusedPulldom::test_dtd_with_external_ref PASSED          [ 62%]
tests.py::TestDefusedPulldom::test_entities_forbidden PASSED             [ 63%]
tests.py::TestDefusedPulldom::test_entity_cycle PASSED                   [ 64%]
tests.py::TestDefusedPulldom::test_external_file_ref PASSED              [ 65%]
tests.py::TestDefusedPulldom::test_external_ref PASSED                   [ 67%]
tests.py::TestDefusedPulldom::test_simple_parse PASSED                   [ 68%]
tests.py::TestDefusedPulldom::test_simple_parse_ns PASSED                [ 69%]
tests.py::TestDefusedSax::test_allow_expansion PASSED                    [ 70%]
tests.py::TestDefusedSax::test_dtd_forbidden PASSED                      [ 71%]
tests.py::TestDefusedSax::test_dtd_with_external_ref PASSED              [ 72%]
tests.py::TestDefusedSax::test_entities_forbidden PASSED                 [ 73%]
tests.py::TestDefusedSax::test_entity_cycle PASSED                       [ 74%]
tests.py::TestDefusedSax::test_exceptions PASSED                         [ 75%]
tests.py::TestDefusedSax::test_external_file_ref PASSED                  [ 76%]
tests.py::TestDefusedSax::test_external_ref PASSED                       [ 78%]
tests.py::TestDefusedSax::test_simple_parse PASSED                       [ 79%]
tests.py::TestDefusedSax::test_simple_parse_ns PASSED                    [ 80%]
tests.py::TestDefusedLxml::test_allow_expansion FAILED                   [ 81%]
tests.py::TestDefusedLxml::test_dtd_forbidden FAILED                     [ 82%]
tests.py::TestDefusedLxml::test_dtd_with_external_ref FAILED             [ 83%]
tests.py::TestDefusedLxml::test_entities_forbidden FAILED                [ 84%]
tests.py::TestDefusedLxml::test_entity_cycle FAILED                      [ 85%]
tests.py::TestDefusedLxml::test_external_file_ref PASSED                 [ 86%]
tests.py::TestDefusedLxml::test_external_ref PASSED                      [ 87%]
tests.py::TestDefusedLxml::test_lxml_warnings FAILED                     [ 89%]
tests.py::TestDefusedLxml::test_restricted_element1 FAILED               [ 90%]
tests.py::TestDefusedLxml::test_restricted_element2 FAILED               [ 91%]
tests.py::TestDefusedLxml::test_simple_parse FAILED                      [ 92%]
tests.py::TestDefusedLxml::test_simple_parse_ns FAILED                   [ 93%]
tests.py::TestDefusedLxml::test_xpath_injection FAILED                   [ 94%]
tests.py::TestXmlRpc::test_monkeypatch PASSED                            [ 95%]
tests.py::TestXmlRpc::test_xmlrpc PASSED                                 [ 96%]
tests.py::TestDefusedGzip::test_defused_gzip_decode PASSED               [ 97%]
tests.py::TestDefusedGzip::test_defused_gzip_response PASSED             [ 98%]
tests.py::test_main PASSED                                               [100%]

=================================== FAILURES ===================================
________________________ BaseTests.test_allow_expansion ________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 177, in test_allow_expansion
    self.parse(self.xml_bomb2, forbid_entities=False)
AttributeError: 'BaseTests' object has no attribute 'parse'
_________________________ BaseTests.test_dtd_forbidden _________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 130, in test_dtd_forbidden
    self.assertRaises(DTDForbidden, self.parse, self.xml_bomb, forbid_dtd=True)
AttributeError: 'BaseTests' object has no attribute 'parse'
_____________________ BaseTests.test_dtd_with_external_ref _____________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 158, in test_dtd_with_external_ref
    self.parse(self.xml_dtd)
AttributeError: 'BaseTests' object has no attribute 'parse'
______________________ BaseTests.test_entities_forbidden _______________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 109, in test_entities_forbidden
    self.assertRaises(EntitiesForbidden, self.parse, self.xml_bomb)
AttributeError: 'BaseTests' object has no attribute 'parse'
_________________________ BaseTests.test_entity_cycle __________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 127, in test_entity_cycle
    self.assertRaises(self.cyclic_error, self.parse, self.xml_cyclic, forbid_entities=False)
AttributeError: 'BaseTests' object has no attribute 'parse'
_______________________ BaseTests.test_external_file_ref _______________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 173, in test_external_file_ref
    self.external_ref_exception, self.parseString, content, forbid_entities=False
AttributeError: 'BaseTests' object has no attribute 'parseString'
_________________________ BaseTests.test_external_ref __________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 162, in test_external_ref
    self.external_ref_exception, self.parse, self.xml_external, forbid_entities=False
AttributeError: 'BaseTests' object has no attribute 'parse'
_________________________ BaseTests.test_simple_parse __________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 97, in test_simple_parse
    self.parse(self.xml_simple)
AttributeError: 'BaseTests' object has no attribute 'parse'
________________________ BaseTests.test_simple_parse_ns ________________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 103, in test_simple_parse_ns
    self.parse(self.xml_simple_ns)
AttributeError: 'BaseTests' object has no attribute 'parse'
_____________________ TestDefusedElementTree.test_html_arg _____________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 201, in test_html_arg
    with self.assertRaises(DeprecationWarning):
  File "/usr/local/lib/python3.10/unittest/case.py", line 226, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File "/usr/local/lib/python3.10/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: DeprecationWarning not raised
____________________ TestDefusedcElementTree.test_html_arg _____________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 201, in test_html_arg
    with self.assertRaises(DeprecationWarning):
  File "/usr/local/lib/python3.10/unittest/case.py", line 226, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File "/usr/local/lib/python3.10/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: DeprecationWarning not raised
_____________________ TestDefusedLxml.test_allow_expansion _____________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 177, in test_allow_expansion
    self.parse(self.xml_bomb2, forbid_entities=False)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
______________________ TestDefusedLxml.test_dtd_forbidden ______________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 130, in test_dtd_forbidden
    self.assertRaises(DTDForbidden, self.parse, self.xml_bomb, forbid_dtd=True)
  File "/usr/local/lib/python3.10/unittest/case.py", line 738, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/local/lib/python3.10/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
__________________ TestDefusedLxml.test_dtd_with_external_ref __________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 359, in test_dtd_with_external_ref
    self.assertRaises(NotSupportedError, self.parse, self.xml_dtd)
  File "/usr/local/lib/python3.10/unittest/case.py", line 738, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/local/lib/python3.10/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
___________________ TestDefusedLxml.test_entities_forbidden ____________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 356, in test_entities_forbidden
    self.assertRaises(NotSupportedError, self.parse, self.xml_bomb)
  File "/usr/local/lib/python3.10/unittest/case.py", line 738, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/local/lib/python3.10/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
______________________ TestDefusedLxml.test_entity_cycle _______________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 127, in test_entity_cycle
    self.assertRaises(self.cyclic_error, self.parse, self.xml_cyclic, forbid_entities=False)
  File "/usr/local/lib/python3.10/unittest/case.py", line 738, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/local/lib/python3.10/unittest/case.py", line 186, in handle
    raise TypeError('%s() arg 1 must be %s' %
TypeError: assertRaises() arg 1 must be an exception type or tuple of exception types
______________________ TestDefusedLxml.test_lxml_warnings ______________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 432, in test_lxml_warnings
    self.assertTrue(lxml_warnings)
  File "/usr/local/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: None is not true
___________________ TestDefusedLxml.test_restricted_element1 ___________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 369, in test_restricted_element1
    tree = self.module.parse(self.xml_bomb, forbid_dtd=False, forbid_entities=False)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 370, in test_restricted_element1
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
___________________ TestDefusedLxml.test_restricted_element2 ___________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 386, in test_restricted_element2
    tree = self.module.parse(self.xml_bomb2, forbid_dtd=False, forbid_entities=False)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 387, in test_restricted_element2
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
______________________ TestDefusedLxml.test_simple_parse _______________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 97, in test_simple_parse
    self.parse(self.xml_simple)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
_____________________ TestDefusedLxml.test_simple_parse_ns _____________________
Traceback (most recent call last):
  File "/workdir/defusedxml-0.7.1/tests.py", line 341, in parse
    tree = self.module.parse(xmlfile, **kwargs)
AttributeError: 'NoneType' object has no attribute 'parse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 103, in test_simple_parse_ns
    self.parse(self.xml_simple_ns)
  File "/workdir/defusedxml-0.7.1/tests.py", line 342, in parse
    except XMLSyntaxError:
TypeError: catching classes that do not inherit from BaseException is not allowed
_____________________ TestDefusedLxml.test_xpath_injection _____________________
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/workdir/defusedxml-0.7.1/tests.py", line 413, in test_xpath_injection
    root = lxml.fromstring(xml)
AttributeError: 'NoneType' object has no attribute 'fromstring'
=============================== warnings summary ===============================
tests.py::TestDefusedElementTree::test_html_arg
tests.py::TestDefusedcElementTree::test_html_arg
  /workdir/defusedxml-0.7.1/defusedxml/ElementTree.py:97: DeprecationWarning: 'html' keyword argument is no longer supported. Pass in arguments as keyword arguments.
    warnings.warn(

tests.py::test_main
  /workdir/defusedxml-0.7.1/.tox/cpythonlibtest-unit-test-tests/lib/python3.10/site-packages/_pytest/python.py:198: PytestReturnNotNoneWarning: Expected None, but tests.py::test_main returned <unittest.suite.TestSuite tests=[<tests.TestDefusedcElementTree testMethod=test_aliases>, <tests.TestDefusedcElementTree testMethod=test_allow_expansion>, <tests.TestDefusedcElementTree testMethod=test_celementtree_warnings>, <tests.TestDefusedcElementTree testMethod=test_dtd_forbidden>, <tests.TestDefusedcElementTree testMethod=test_dtd_with_external_ref>, <tests.TestDefusedcElementTree testMethod=test_entities_forbidden>, <tests.TestDefusedcElementTree testMethod=test_entity_cycle>, <tests.TestDefusedcElementTree testMethod=test_external_file_ref>, <tests.TestDefusedcElementTree testMethod=test_external_ref>, <tests.TestDefusedcElementTree testMethod=test_html_arg>, <tests.TestDefusedcElementTree testMethod=test_import_order>, <tests.TestDefusedcElementTree testMethod=test_orig_parseerror>, <tests.TestDefusedcElementTree testMethod=test_simple_parse>, <tests.TestDefusedcElementTree testMethod=test_simple_parse_ns>, <tests.TestDefusedElementTree testMethod=test_aliases>, <tests.TestDefusedElementTree testMethod=test_allow_expansion>, <tests.TestDefusedElementTree testMethod=test_dtd_forbidden>, <tests.TestDefusedElementTree testMethod=test_dtd_with_external_ref>, <tests.TestDefusedElementTree testMethod=test_entities_forbidden>, <tests.TestDefusedElementTree testMethod=test_entity_cycle>, <tests.TestDefusedElementTree testMethod=test_external_file_ref>, <tests.TestDefusedElementTree testMethod=test_external_ref>, <tests.TestDefusedElementTree testMethod=test_html_arg>, <tests.TestDefusedElementTree testMethod=test_import_order>, <tests.TestDefusedElementTree testMethod=test_orig_parseerror>, <tests.TestDefusedElementTree testMethod=test_simple_parse>, <tests.TestDefusedElementTree testMethod=test_simple_parse_ns>, <tests.TestDefusedMinidom testMethod=test_allow_expansion>, <tests.TestDefusedMinidom testMethod=test_dtd_forbidden>, <tests.TestDefusedMinidom testMethod=test_dtd_with_external_ref>, <tests.TestDefusedMinidom testMethod=test_entities_forbidden>, <tests.TestDefusedMinidom testMethod=test_entity_cycle>, <tests.TestDefusedMinidom testMethod=test_external_file_ref>, <tests.TestDefusedMinidom testMethod=test_external_ref>, <tests.TestDefusedMinidom testMethod=test_simple_parse>, <tests.TestDefusedMinidom testMethod=test_simple_parse_ns>, <tests.TestDefusedMinidomWithParser testMethod=test_allow_expansion>, <tests.TestDefusedMinidomWithParser testMethod=test_dtd_forbidden>, <tests.TestDefusedMinidomWithParser testMethod=test_dtd_with_external_ref>, <tests.TestDefusedMinidomWithParser testMethod=test_entities_forbidden>, <tests.TestDefusedMinidomWithParser testMethod=test_entity_cycle>, <tests.TestDefusedMinidomWithParser testMethod=test_external_file_ref>, <tests.TestDefusedMinidomWithParser testMethod=test_external_ref>, <tests.TestDefusedMinidomWithParser testMethod=test_simple_parse>, <tests.TestDefusedMinidomWithParser testMethod=test_simple_parse_ns>, <tests.TestDefusedPulldom testMethod=test_allow_expansion>, <tests.TestDefusedPulldom testMethod=test_dtd_forbidden>, <tests.TestDefusedPulldom testMethod=test_dtd_with_external_ref>, <tests.TestDefusedPulldom testMethod=test_entities_forbidden>, <tests.TestDefusedPulldom testMethod=test_entity_cycle>, <tests.TestDefusedPulldom testMethod=test_external_file_ref>, <tests.TestDefusedPulldom testMethod=test_external_ref>, <tests.TestDefusedPulldom testMethod=test_simple_parse>, <tests.TestDefusedPulldom testMethod=test_simple_parse_ns>, <tests.TestDefusedSax testMethod=test_allow_expansion>, <tests.TestDefusedSax testMethod=test_dtd_forbidden>, <tests.TestDefusedSax testMethod=test_dtd_with_external_ref>, <tests.TestDefusedSax testMethod=test_entities_forbidden>, <tests.TestDefusedSax testMethod=test_entity_cycle>, <tests.TestDefusedSax testMethod=test_exceptions>, <tests.TestDefusedSax testMethod=test_external_file_ref>, <tests.TestDefusedSax testMethod=test_external_ref>, <tests.TestDefusedSax testMethod=test_simple_parse>, <tests.TestDefusedSax testMethod=test_simple_parse_ns>, <tests.TestXmlRpc testMethod=test_monkeypatch>, <tests.TestXmlRpc testMethod=test_xmlrpc>, <tests.TestDefusedGzip testMethod=test_defused_gzip_decode>, <tests.TestDefusedGzip testMethod=test_defused_gzip_response>]>, which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /workdir/results/defusedxml/0.7.1/1/cpython-test-results.xml -
=========================== short test summary info ============================
FAILED tests.py::BaseTests::test_allow_expansion - AttributeError: 'BaseTests...
FAILED tests.py::BaseTests::test_dtd_forbidden - AttributeError: 'BaseTests' ...
FAILED tests.py::BaseTests::test_dtd_with_external_ref - AttributeError: 'Bas...
FAILED tests.py::BaseTests::test_entities_forbidden - AttributeError: 'BaseTe...
FAILED tests.py::BaseTests::test_entity_cycle - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_external_file_ref - AttributeError: 'BaseTes...
FAILED tests.py::BaseTests::test_external_ref - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse_ns - AttributeError: 'BaseTests...
FAILED tests.py::TestDefusedElementTree::test_html_arg - AssertionError: Depr...
FAILED tests.py::TestDefusedcElementTree::test_html_arg - AssertionError: Dep...
FAILED tests.py::TestDefusedLxml::test_allow_expansion - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_dtd_forbidden - TypeError: catching cl...
FAILED tests.py::TestDefusedLxml::test_dtd_with_external_ref - TypeError: cat...
FAILED tests.py::TestDefusedLxml::test_entities_forbidden - TypeError: catchi...
FAILED tests.py::TestDefusedLxml::test_entity_cycle - TypeError: assertRaises...
FAILED tests.py::TestDefusedLxml::test_lxml_warnings - AssertionError: None i...
FAILED tests.py::TestDefusedLxml::test_restricted_element1 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_restricted_element2 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_simple_parse - TypeError: catching cla...
FAILED tests.py::TestDefusedLxml::test_simple_parse_ns - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_xpath_injection - AttributeError: 'Non...
================== 22 failed, 69 passed, 3 warnings in 0.23s ===================
ERROR: InvocationError for command /workdir/defusedxml-0.7.1/.tox/cpythonlibtest-unit-test-tests/bin/pytest -v --tb=native --junitxml /workdir/results/defusedxml/0.7.1/1/cpython-test-results.xml (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   cpythonlibtest-unit-test-tests: commands failed
